src/datachain/lib/dc.py

```diff
@@ -1181,9 +1181,15 @@ def to_pandas(self, flatten=False) -> "pd.DataFrame":
         """
         headers, max_length = self._effective_signals_schema.get_headers_with_length()
         if flatten or max_length < 2:
-            df = pd.DataFrame.from_records(self.to_records())
-            if headers:
-                df.columns = [".".join(filter(None, header)) for header in headers]
+            columns = [".".join(filter(None, header)) for header in headers]
```

Comment:
see the fix / implementation in https://github.com/iterative/datachain/pull/241/files - seems a bit simpler (+adds columns in the nested case below for consistency)